### PR TITLE
PoC - GetProperties function for untyped objects

### PR DESCRIPTION
**Pull request by CarlosFigueiraMSFT**
Alternative implementation: #2100

### CODE REVIEW

File: src/strings/PowerFxResources.en-US.resx [hunk 1]
```diff
@@ -1704,6 +1704,18 @@
     <value>separator</value>
     <comment>function_parameter - Second argument to the Split function - the delimiter text that is used to split SplitArg1 into substrings.</comment>
   </data>
+  <data name="AboutGetProperties" xml:space="preserve">
+    <value>Enumerate and returns the properties of an untyped record.</value>
+    <comment>Description of 'GetProperties' function. The term 'record' is used in the context of an obejct / property bag / set of name/value pairs.</comment>
```
Comment (jorisdg) [inline src/strings/PowerFxResources.en-US.resx:1709]: typoe: obejct

File: src/strings/PowerFxResources.en-US.resx [hunk 2]
```diff
@@ -1704,6 +1704,18 @@
...
+  </data>
+  <data name="GetPropertiesArg1" xml:space="preserve">
+    <value>record</value>
+    <comment>function_parameter - First argument to the GetProperties function - the record whose inputs will be returned. The term 'record' is used in the context of an obejct / property bag / set of name/value pairs.</comment>
```
Comment (jorisdg) [inline src/strings/PowerFxResources.en-US.resx:1713]: obejct

File: src/strings/PowerFxResources.en-US.resx [hunk 3]
```diff
@@ -1704,6 +1704,18 @@
...
+  </data>
+  <data name="AboutGetProperties_record" xml:space="preserve">
+    <value>Record to have its properties enumerated.</value>
+    <comment>Description of first argument of the 'GetProperties' function. The term 'record' is used in the context of an obejct / property bag / set of name/value pairs.</comment>
```
Comment (jorisdg) [inline src/strings/PowerFxResources.en-US.resx:1717]: copy paste :D

File: src/strings/PowerFxResources.en-US.resx (hunk 1 above)
Comment (CarlosFigueiraMSFT) [reply src/strings/PowerFxResources.en-US.resx]: Updated, thanks!

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/GetProperties.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    // GetProperties(record:O): *[Name:s,Value:O]
+    internal class GetPropertiesFunction_UO : BuiltinFunction
+    {
+        public override bool IsSelfContained => true;
+
+        internal const string NameColumnName = "Name";
+        internal const string ValueColumnName = "Value";
```
Comment (jas-valgotar) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/GetProperties.cs:21]: >internal const string ValueColumnName = "Value"; [](http://example.com/codeflow?start=8&length=48)

Can we reuse `ColumnName_ValueStr` from TexlFunction?

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by CarlosFigueiraMSFT** [comment]
Closed as it was implemented in #2100

**Pull request closed without merging**